test/extended/devices/recovery.go

```diff
@@ -0,0 +1,416 @@
+							MountPath: "/var/lib/kubelet/plugins_registry",
+						},
+						{
+							Name:      "dev",
+							MountPath: "/dev",
+						},
+					},
+				},
+			},
+		},
+	}
+	return podDef
+}
+
+func waitForPodRunningOrFail(k8sCli kubernetes.Interface, podNamespace, podName string) *corev1.Pod {
+	var updatedPod *corev1.Pod
+	var err error
+	// timeout is generous because we need to handle the challenging post-reboot environment
+	o.Eventually(func() (bool, error) {
+		updatedPod, err = k8sCli.CoreV1().Pods(podNamespace).Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return exutil.CheckPodIsRunning(*updatedPod), nil
+	}, 5*time.Minute, 2*time.Second).Should(o.BeTrue(), "pod %s/%s never got running", podNamespace, podName)
+	return updatedPod
+}
+
+func makeWorkloadPod(namespace, SampleDeviceResourceName, cmd string) *corev1.Pod {
+	zero := int64(0)
+	rl := corev1.ResourceList{
+		corev1.ResourceName(SampleDeviceResourceName): *resource.NewQuantity(1, resource.DecimalSI),
+	}
+	podDef := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "sample-device-workload-",
+			Namespace:    namespace,
+			Labels:       workloadPodLabel,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy:                 corev1.RestartPolicyAlways,
+			TerminationGracePeriodSeconds: &zero,
+			SecurityContext: &corev1.PodSecurityContext{
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: corev1.SeccompProfileTypeRuntimeDefault,
+				},
+			},
+			Containers: []corev1.Container{{
+				Image: image.ShellImage(),
+				Name:  "sample-device-workload",
+				Command: []string{
+					"sh", "-c", cmd,
+				},
+				Resources: v1.ResourceRequirements{
+					Limits:   rl,
+					Requests: rl,
+				},
+			}},
+		},
+	}
+	return podDef
+}
+
+func getPodLogs(oc *exutil.CLI, namespace, name string) (string, error) {
+	byteOutput, err := oc.AsAdmin().KubeClient().CoreV1().Pods(namespace).GetLogs(name, &corev1.PodLogOptions{}).DoRaw(context.Background())
+	return string(byteOutput), err
+}
+
+func prepareRegistrationControlOnNodeOrFail(oc *exutil.CLI, node *corev1.Node) {
+	framework.Logf("Preparing node %q for sample device plugin registration", node.Name)
+	fwk := oc.KubeFramework()
+	_, err := exutil.ExecCommandOnMachineConfigDaemon(fwk.ClientSet, oc, node, []string{
+		"sh", "-c", setupRegistrationCommandMCD,
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	framework.Logf("Prepared node %q for sample device plugin registration", node.Name)
+}
+
+func unblockRegistrationControlOnNodeOrFail(oc *exutil.CLI, node *corev1.Node) {
+	framework.Logf("Preparing node %q for sample device plugin registration", node.Name)
+	fwk := oc.KubeFramework()
+	_, err := exutil.ExecCommandOnMachineConfigDaemon(fwk.ClientSet, oc, node, []string{
+		"sh", "-c", unblockRegistrationCommandMCD,
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	framework.Logf("Prepared node %q for sample device plugin registration", node.Name)
```

Comment:
NIT: Log messages here need to be updated to capture that they for unblocking the device plugin registration process.

Reply:
good point. Will add to the existing log.

Reply:
my fault, I misread. Fixed the log entry.